packages/compose/src/hooks/use-dragging/index.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useCallback, useState, useRef } from '@wordpress/element';
+
+export default function useDragging( { onDragStart, onDragMove, onDragEnd } ) {
+	const [ isDragging, setIsDragging ] = useState( false );
+
+	const eventsRef = useRef( {
+		onDragStart,
+		onDragMove,
+		onDragEnd,
+	} );
+	useEffect(
```

Comment:
We should use `useIsomorphicLayoutEffect` for updating refs like this.

Reply:
Hi @epiqueras, in this case, the refs we are updating do not contain dom nodes they just contain functions and don't depend on the dom at all. Should we still use useIsomorphicLayoutEffect? Do you a link where I can learn useIsomorphicLayoutEffect in more detail so I can avoid this problem in the future? Thank you for any insights you may provide!

Reply:
Because `useEffect` is not synchronous, you can get a rerender with handler changes at the same time an event fires. Then you are leaving it up to the scheduler to decide whether to update your handler refs first or handle the event first, effectively allowing for the possibility of an event being handled with a stale handler.